index.d.ts

```diff
@@ -0,0 +1,86 @@
+declare module 'tuyapi' {
+    import { EventEmitter } from 'events';
+
+    interface TuyaDeviceOptions {
+        ip?: string;
+        port?: number;
+        id: string;
+        gwID?: string;
+        key: string;
+        productKey?: string;
+        version?: number|string;
+        nullPayloadOnJSONError?: boolean;
+        issueGetOnConnect?: boolean;
+        issueRefreshOnConnect?: boolean;
+        issueRefreshOnPing?: boolean;
+    }
+
+    type UnionTypes = Object|number|string|boolean;
+
+    interface Object {
```

Comment:
This could probably just be `Record<string, unknown>`

Reply:
I think the `any` type might be a better fit in this case
![](https://ioexception.at/f9sBkst97L.png)

Reply:
I would rather have `unknown` to force consumers to type the expected payload themselves (since it varies so much from device-to-device) but `any` is fine for now. 😃